### PR TITLE
feat(core): add management APIs to help with linking social identities to user

### PR DESCRIPTION
**Pull request by charIeszhao**
## Summary

Provide new management APIs to help with linking social identities to a given user. This feature is requested by our community and will allow them to implement link social accounts feature in their user profile page (similar to what we have in the cloud admin console).

- POST `/connectors/:connectorId/authorization-uri`
- POST `/users/:userId/identities`

## Testing

Locally tested with postman.

## Checklist

- [x] `.changeset`
- [x] unit tests
- [x] integration tests
- [ ] docs

### CODE REVIEW

File: packages/core/src/routes/connector/authorization-uri.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import assertThat from '#src/utils/assert-that.js';
+
+import { type AuthedRouter, type RouterInitArgs } from '../types.js';
+
+export default function connectorAuthorizationUriRoutes<T extends AuthedRouter>(
+  ...[router, tenant]: RouterInitArgs<T>
+) {
+  const { getLogtoConnectorById } = tenant.connectors;
+
+  router.post(
```
Comment (wangsijie) [inline packages/core/src/routes/connector/authorization-uri.ts:15]: Consider using "GET"?

**Review by wangsijie — APPROVED** [review]

File: packages/core/src/routes/admin-user-social.test.ts [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+const adminUserSocialRoutes = await pickDefault(import('./admin-user-social.js'));
+
+describe('Admin user social identities APIs', () => {
+  const tenantContext = new MockTenant(undefined, mockedQueries, mockedConnectors, {
+    users: usersLibraries,
+  });
+
+  const userRequest = createRequester({ authedRoutes: adminUserSocialRoutes, tenantContext });
+
+  describe('/users/:userId/identities', () => {
```
[the remark this answers is not in the document]
Comment (charIeszhao) [reply packages/core/src/routes/admin-user-social.test.ts]: updated

File: packages/core/src/routes/connector/authorization-uri.ts (hunk 1 above)
Comment (wangsijie) [reply packages/core/src/routes/connector/authorization-uri.ts:15]: OK

File: packages/core/src/routes/admin-user-social.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  router.post(
+    '/users/:userId/identities',
+    koaGuard({
+      params: object({ userId: string() }),
+      body: object({
+        connectorId: string(),
+        connectorData: record(string(), unknown()),
+      }),
+      status: [204, 404, 422],
+    }),
```
Comment (darcyYe) [inline packages/core/src/routes/admin-user-social.ts:31]: should we also return the user info after updating its `identities`?

**Review by darcyYe — APPROVED** [review]
Overall LGTM

File: packages/core/src/routes/admin-user-social.ts (hunk 3 above)
Comment (charIeszhao) [reply packages/core/src/routes/admin-user-social.ts:31]: I'm not sure. Initially I thought it was unnecessary to do so, but now I think maybe you are right...

Comment (charIeszhao) [reply packages/core/src/routes/admin-user-social.ts:37]: Updated

**Review by gao-sun — APPROVED** [review]

**Pull request merged**